### PR TITLE
feat(core): add adapter for boolean primitive data type

### PR DESCRIPTION
**Pull request by ngouy**
## What is this about
Introduce boolean primitive data type  adapter

### CODE REVIEW

File: packages/core/r4b/data-types/boolean.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+      const sanitizedValue = value?.trim()?.toLowerCase();
+      if (!sanitizedValue) {
+        return undefined;
+      }
+
+      // value is a string
+      if (!sanitizedValue.match(fhirBooleanRegexp))
+        throw new Error(
+          "Value does not match the fhir boolean format as described in `https://hl7.org/fhir/datatypes.html#boolean'"
```
Comment (julienblin) [inline packages/core/r4b/data-types/boolean.ts]:
```suggestion
          "Value does not match the fhir boolean format as described in `https://hl7.org/fhir/datatypes.html#boolean`
```

File: packages/core/r4b/data-types/boolean.ts [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      switch (fhirBoolean) {
+        case true:
+          return labels["true"];
+        case false:
+          return labels["false"];
+        case null:
+          return labels["null"];
+        case undefined:
+          return labels["undefined"];
+      }
```
Comment (julienblin) [inline packages/core/r4b/data-types/boolean.ts]: Javascript is happy to coerce strings when using templates :-) 

```suggestion
      return labels[`${fhirBoolean}`];
```

File: packages/core/r4b/data-types/boolean.ts [hunk 3]
```diff
@@ -0,0 +1,88 @@
+import { merge } from "lodash";
```
Comment (julienblin) [inline packages/core/r4b/data-types/boolean.ts]: This one is tricky - lodash is not an esm module, so to ensure compatiblity you need to import it from the root only.

```suggestion
import _ from "lodash";
```

File: packages/core/r4b/data-types/boolean.ts [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+      return sanitizedValue === "true";
+    },
+
+    format(value, options) {
+      const fhirBoolean =
+        typeof value === "boolean" ? value : this.parse(value);
+
+      options ||= {};
+      const labels = merge({}, defaultLabels, options.labels || {});
```
Comment (julienblin) [inline packages/core/r4b/data-types/boolean.ts]:
```suggestion
      const labels = _.merge({}, defaultLabels, options.labels || {});
```

File: packages/core/r4b/data-types/boolean.ts [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+ *
+ * @see https://hl7.org/fhir/datatypes.html#boolean
+ */
+
+export interface FhirBooleanFormatOptions {
+  labels?: {
+    true?: string;
+    false?: string;
+    null?: string;
+    undefined?: string;
```
Comment (julienblin) [inline packages/core/r4b/data-types/boolean.ts]: Generally, `null` and `undefined` are functionally equivalent.
So we could group them as `nil` to make it easier?

```suggestion
    nil?: string;
```

**Review by julienblin — CHANGES_REQUESTED** [review]
A few nitpicks.

Also for options, just a quick remark: I believe `null` and `undefined` are semantically equivalent for a formatter, so we could group `null` and `undefined` labels as `nil`.

File: packages/core/r4b/data-types/boolean.ts (hunk 2 above)
Comment (ngouy) [reply packages/core/r4b/data-types/boolean.ts]: ho Not sure if TS will be happy

I was using `String(fhirBoolean)` and he was not happy

**Comment by ngouy** [comment]
@julienblin that would be the last one of the primitives types

**Review by julienblin — APPROVED** [review]
LGTM!

Sorry for the delay - I was looking at a stale version of the branch, so I was not looking at your latest changes.

**Pull request merged**